src/Nethermind/Nethermind.Consensus.AuRa/Transactions/RandomContractTxSource.cs

```diff
@@ -54,7 +54,7 @@ public class RandomContractTxSource : ITxSource
         
         public IEnumerable<Transaction> GetTransactions(BlockHeader parent, long gasLimit)
         {
-            if (_contracts.TryGetForBlock(parent.Number + 1, out var contract))
+            if (_contracts.TryGetForActivation(parent.Number + 1, out var contract))
```

Comment:
there is this very strange thing when the activation can be both block number and time but then you totally break Liskov substitution and use the interface and pass the block number?